packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py

```diff
@@ -75,42 +78,37 @@ async def tasks_manager() -> AsyncIterator[TasksManager]:
     await tasks_manager.close()
 
 
-async def test_unchecked_task_is_auto_removed(tasks_manager: TasksManager):
+@pytest.mark.parametrize("check_task_presence_before", [True, False])
+async def test_task_is_auto_removed(
+    tasks_manager: TasksManager, check_task_presence_before: bool
+):
     task_id = start_task(
         tasks_manager,
         a_background_task,
         raise_when_finished=False,
         total_sleep=10 * TEST_CHECK_STALE_INTERVAL_S,
     )
-    await asyncio.sleep(2 * TEST_CHECK_STALE_INTERVAL_S + 1)
-    async for attempt in AsyncRetrying(**_RETRY_PARAMS):
-        with attempt:
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_status(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result_old(task_id)
 
+    if check_task_presence_before:
+        # immediately after starting the task is still there
+        task_status = tasks_manager.get_task_status(task_id, with_task_context=None)
+        assert task_status
 
-async def test_checked_once_task_is_auto_removed(tasks_manager: TasksManager):
-    task_id = start_task(
-        tasks_manager,
-        a_background_task,
-        raise_when_finished=False,
-        total_sleep=10 * TEST_CHECK_STALE_INTERVAL_S,
-    )
-    # check once (different branch in code)
-    tasks_manager.get_task_status(task_id, with_task_context=None)
-    await asyncio.sleep(2 * TEST_CHECK_STALE_INTERVAL_S + 1)
+    # wait for task to be automatically removed
+    # meaning no calls via the manager methods are received
     async for attempt in AsyncRetrying(**_RETRY_PARAMS):
         with attempt:
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_status(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result(task_id, with_task_context=None)
-            with pytest.raises(TaskNotFoundError):
-                tasks_manager.get_task_result_old(task_id)
+            for tasks in tasks_manager._tasks_groups.values():  # noqa: SLF001
+                if task_id in tasks:
+                    msg = "wait till no element is found any longer"
+                    raise AssertionError(msg)
```

Comment:
MINOR: if you want to make it more readable, you can use `raise tenacity.TryAgain(msg)`